plotly/offline/offline.py

```diff
@@ -236,13 +247,39 @@ def iplot(figure_or_data, show_link=True, link_text='Export to plot.ly',
 
     display(HTML(plot_html))
 
+    if image:
+        warnings.warn('For higher resolution images and more export '
+                      'options, consider making requests to the '
+                      'Plotly image servers. Type help(\'py.image\') '
+                      'for more details.', UserWarning)
+        # Write script to download image of the plot
+        script = ('<script>'
+                  'function downloadimage(format, height, width,'
+                  ' filename) {{'
+                  'var p = document.getElementById(\'{plot_id}\');'
+                  'Plotly.downloadImage(p, {{format: format, height: height, '
+                  'width: width, filename: filename}});'
+                  '}};'
+                  'if(document.readyState == \'complete\') {{'
+                  'downloadimage(\'{format}\', {height}, {width}, '
+                  '\'{filename}\');}}'
+                  '</script>'
+                  ).format(format=format, width=_width, height=_height,
+                           filename=filename, plot_id=plotdivid)
+        # allow time for the plot to draw
+        import time
+        time.sleep(1)
```

Comment:
Right so this is a bit of a concern. The most complicated plot I tried was one with 300 boxplots, I also dropped `time.sleep` to `0.1` and there was no issue. I'm not really sure of a way to guarantee this success though.